### PR TITLE
Directions endpoint and filter bus stops by direction

### PR DESCRIPTION
**Pull request by jtrudell**
- Adds bus_routes/id/directions endpoint and directions controller
- Adds by_direction scope to BusStop model
- If params[:direction], BusStop controller will filter by direction
- Tests

### CODE REVIEW

File: app/models/bus_stop.rb [hunk 1]
```diff
@@ -3,6 +3,8 @@ class BusStop < ApplicationRecord
   validates_presence_of :stop_id, :name, :direction
   validates_uniqueness_of :stop_id
 
+  scope :by_direction, ->(direction) { where('direction = ?', direction) }
```
Comment (erinnachen) [inline app/models/bus_stop.rb]: `where(direction: direction)`

**Review by erinnachen — COMMENTED** [review]
For bus_stops#index and directions#index, these are not setup to respond with json. You'll need either a separate endpoint (api/bus_routes) or if you want to share the endpoint, something like this: http://guides.rubyonrails.org/action_controller_overview.html#rendering-xml-and-json-data

**Comment by jtrudell** [comment]
@erinnachen fair enough

**Comment by jtrudell** [comment]
@erinnachen This is where the react-with-rails thing starts to throw me off. You sometimes use the views and #to_json the data, although there you are using the html id tag as a hook for the react component. But sometimes you just want JSON, because you aren't rendering a component. Makes my head hurt.

File: test/models/bus_route_test.rb [hunk 2]
```diff
@@ -32,29 +32,37 @@ class BusRouteTest < ActiveSupport::TestCase
   end
 
   test 'invalid if external_id not uniuqe' do
-    bus_route = BusRoute.create(external_id: '555', name: 'My Aweomse Bus')
+    BusRoute.create(external_id: '555', name: 'My Aweomse Bus')
```
Comment (erinnachen) [inline test/models/bus_route_test.rb]: awesome

File: test/models/bus_route_test.rb [hunk 3]
```diff
@@ -32,29 +32,37 @@ class BusRouteTest < ActiveSupport::TestCase
...
     bus_route = BusRoute.new(external_id: '555', name: 'Party Bus')
     assert bus_route.invalid?
-    assert_equal ["has already been taken"], bus_route.errors[:external_id]
+    assert_equal ['has already been taken'], bus_route.errors[:external_id]
+  end
+
+  test 'has many directions' do
+    bus_route = BusRoute.create(external_id: '555', name: 'My Aweomse Bus')
```
Comment (erinnachen) [inline test/models/bus_route_test.rb]: wooo

File: test/models/bus_stop_test.rb [hunk 4]
```diff
@@ -1,32 +1,32 @@
...
-    assert_equal "Ashland & Blackhawk", bus_stop.name
-    assert_equal "Westbound", bus_stop.direction
+    assert_equal '333', bus_stop.stop_id
+    assert_equal 'Ashland & Blackhawk', bus_stop.name
+    assert_equal 'Westbound', bus_stop.direction
   end
 
   test 'valid bus stop' do
-    bus_stop = BusStop.new(stop_id: '555', name: 'Fith and Third', direction: "Northbound")
+    bus_stop = BusStop.new(stop_id: '555', name: 'Fith and Third', direction: 'Northbound')
```
Comment (erinnachen) [inline test/models/bus_stop_test.rb]: fifth sp

File: test/models/bus_stop_test.rb [hunk 5]
```diff
@@ -38,9 +38,16 @@ class BusStopTest < ActiveSupport::TestCase
   end
 
   test 'invalid if stop_id not uniuqe' do
```
Comment (erinnachen) [inline test/models/bus_stop_test.rb]: unique

File: test/models/bus_stop_test.rb [hunk 6]
```diff
@@ -38,9 +38,16 @@ class BusStopTest < ActiveSupport::TestCase
...
+    bus_stop = BusStop.new(stop_id: '555', name: 'Foo Bar', direction: 'Southbound')
     assert bus_stop.invalid?
-    assert_equal ["has already been taken"], bus_stop.errors[:stop_id]
+    assert_equal ['has already been taken'], bus_stop.errors[:stop_id]
+  end
+
+  test 'scope by_direction' do
+    BusStop.create(stop_id: '555', name: 'Fith and Third', direction: 'Northbound')
+    bus_stop = BusStop.create(stop_id: '666', name: 'Foo Bar', direction: 'Southbound')
+    assert_equal 1, BusStop.all.by_direction('Southbound').count
```
Comment (erinnachen) [inline test/models/bus_stop_test.rb]: should not need the `.all` scope here

**Review by erinnachen — COMMENTED** [review]
My job is to spell check.

**Comment by jtrudell** [comment]
@erinnachen done. or "dun".

**Review by erinnachen — APPROVED** [review]

**Pull request merged**